File: Programmers/C++/Lv1_콜라츠_추측.cpp
#include <string>
#include <vector>

using namespace std;

int solution(int num) {
    int answer = 0;
    long long n = num;
    
    while (1) {
        if (n == 1)
            break;
        
        if (n % 2 == 0)
            n /= 2;
        else
            n = n * 3 + 1;
        answer++;
        
        if (answer == 500)
            return -1;
    }
    return answer;
}

File: Programmers/Python/Level_1/Lv1_콜라츠_추측.py
def solution(num):
    answer = 0
    
    while True:
        if num == 1:
            break
        if num % 2 == 0:
            num /= 2
        elif num % 2 == 1:
            num = num*3+1
        answer += 1
        if answer == 500:
            return -1
    return answer